src/qt/veil/miningwidget.cpp

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2021 The Veil developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <qt/veil/miningwidget.h>
+#include <qt/veil/forms/ui_miningwidget.h>
+
+// SOME OF THESE NEED TO BE DELETED
+#include <qt/guiconstants.h>
+#include <qt/guiutil.h>
+#include <qt/optionsmodel.h>
+#include <qt/platformstyle.h>
+#include <interfaces/node.h>
+#include <qt/walletview.h>
+#include <qt/walletmodel.h>
+#include <key_io.h>
+#include <wallet/wallet.h>
+#include <QGraphicsOpacityEffect>
+#include <QPropertyAnimation>
+#include <QPixmap>
+#include <QIcon>
+#include <QClipboard>
+#include <QMimeData>
+#include <QImage>
+#include <QLabel>
+#include <QPainter>
+#include <qt/veil/qtutils.h>
+#include <miner.h>
+
+MiningWidget::MiningWidget(QWidget *parent, WalletView* walletView) :
+    QWidget(parent),
+    ui(new Ui::MiningWidget),
+    mainWindow(walletView) {
+    ui->setupUi(this);
+    ui->title->setProperty("cssClass" , "title");
+
+    // SET THE NUMBER OF THREADS AND ALGORITHM HERE
+    mineOn = GenerateActive();
+
+    ui->cmbAlgoSelect->addItems({"RandomX","ProgPow","SHA256D"});
+    ui->cmbAlgoSelect->setCurrentIndex(GetMiningAlgorithm());
+
+    connect(ui->btnUpdateAlgo, SIGNAL(clicked()), this, SLOT(onUpdateAlgorithm()));
+    connect(ui->btnAllThreads, SIGNAL(clicked()), this, SLOT(onUseMaxThreads()));
+    connect(ui->btnActiveMine, SIGNAL(clicked()), this, SLOT(onToggleMiningActive()));
+
+}
+
+void MiningWidget::setMineActiveTxt(bool mineActive) {
+    ui->lblMineActive->setText(mineActive ? "Mining Active" : "Mining Stopped");
+    ui->lblMineActive->setStyleSheet(mineActive ? "QLabel{color:green;}" : "QLabel{color:red;}");
+    ui->btnActiveMine->setText(mineActive ? "Stop Mining" : "Start Mining");
+}
+
+void MiningWidget::setWalletModel(WalletModel *model) {
+    this->walletModel = model;
+
+    connect(model, SIGNAL(updateMiningFields()), this, SLOT(updateMiningFields()));
+}
+
+MiningWidget::~MiningWidget()
+{
+    delete ui;
+}
+
+void MiningWidget::onUpdateAlgorithm() {
+    if (mineOn) {
+        openToastDialog(QString::fromStdString("Mining must be stopped to change algorithm!"), mainWindow->getGUI());
+    } else {
+        currentMiningAlgo = ui->cmbAlgoSelect->currentIndex(); 
+        std::string algoStr = GetMiningType(currentMiningAlgo, false, false);
+        bool setAlgoResult = SetMiningAlgorithm(algoStr); 
+
+        if (setAlgoResult) {
+            openToastDialog(QString::fromStdString("Aglorithm Switch Success!"), mainWindow->getGUI());
```

Comment:
nit: typo

Reply:
Since this is on a message presented to the user, and not a rare message; I'm going to upgrade this above just a 'nit'.